google-oauth-client/src/main/java/com/google/api/client/auth/oauth/OAuthHmacSha256Signer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.auth.oauth;
+
+import com.google.api.client.util.Base64;
+import com.google.api.client.util.Beta;
+import com.google.api.client.util.StringUtils;
+import java.security.GeneralSecurityException;
+import javax.crypto.Mac;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * {@link Beta} <br>
+ * OAuth {@code "HMAC-SHA256"} signature method.
+ */
+@Beta
+public final class OAuthHmacSha256Signer implements OAuthSigner {
+
+  /** Client secret */
+  private String clientSharedSecret;
+
+  /** Token secret */
+  private String tokenSharedSecret;
+
+  public void setClientSecret(String clientSecret) {
```

Comment:
Why are these mutable? It makes the class not thread-safe.

Reply:
The client secret can be final, so I did change it, however the token secret usually is not known until after the first request of the three-legged OAuth 1.0 authorization flow.  Because of that, it needs to be able to be set later.  This is common for OAuth 1.0, as it is a multi-step process where not all of the values are known prior to the first step.

The token secret will usually be set to the final value only once during the authorization flow, before the second step, and never changed again.